modules/creating-manifest-file-customized-br-ex-bridge.adoc

```diff
@@ -0,0 +1,193 @@
+endif::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+
+.Prerequisites
+
+
+.Procedure
+ifndef::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+. Create a `MachineConfig` manifest file and define a customized `br-ex` bridge network configuration similiar to the following example:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker <1>
+  name: 10-br-ex-worker
+spec:
+nodeSelector:
+    kubernetes.io/hostname: worker-0
+    desiredState:
+    interfaces:
+    - name: enp2s0 <2>
+      type: ethernet <3>
+      state: up <4>
+      ipv4:
+        enabled: false <5>
+      ipv6:
+        enabled: false
+    - name: br-ex
+      type: ovs-bridge
+      state: up
+      ipv4:
+         enabled: false
+         dhcp: false
+      ipv6:
+         enabled: false
+         dhcp: false
+      bridge:
+       port:
+       - name: enp2s0 <6>
+       - name: br-ex
+    - name: br-ex
+      type: ovs-interface
+      state: up
+      copy-mac-from: enp2s0
+      ipv4:
+        enabled: true
+        dhcp: true
+      ipv6:
+        enabled: false
+        dhcp: false
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - contents: <7>
+          source: <bse64_encoded_access_key_id>
+          mode: 0644
+          overwrite: true
+          path: /etc/nmstate/openshift/<hostname>.yml
+# ...
+----
+<1> Specify the `NetworkManager` connection name for the primary network interface.
+<2> Name of the interface.
+<3> The type of interface. This example creates a bridge.
+<4> The requested state for the interface after creation.
+<5> Disables IPv4 and IPv6 in this example.
+<6> The node NIC to which the bridge is attached.
+<7> For each node in your cluster, specify the hostname path to your node and the base-64 encoded Ignition configuration file data for the machine type. If you have a single global configuration specified in an `/etc/nmstate/openshift/cluster.yml` configuration file that you want to apply to all nodes in your cluster, you do not need to specify the hostname path for each node.
++
+[NOTE]
+====
+The `.yaml` extension does not work when specifying the hostname path for each node or all nodes in the `MachineConfig` manifest file.
+====
+endif::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+ifdef::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+. Create a `NodeNetworkConfigurationPolicy (NNCP)` manifest file and define a customized `br-ex` bridge network configuration similiar to the following example:
```

Comment:
We should note that this should be the same as what was specified on day 1, with the addition of the masquerade IP (the 169.254.169.2 address in the example).

We also need to note that the masquerade IP depends on the IP version in use. The 169 address is for ipv4, but on ipv6 it's fd69::2/128 (I think, I need to verify the netmask).

Reply:
Updated step to state:

`Depending on your needs, ensure that you set a masquerade IP for either the `ipv4.address.ip`, `ipv6.address.ip`, or both parameters.`

For `ipv6`, do I ned to include a `prefix-length` parameter? If so, what value should I state? Is `29` OK for `ipv4`?

![Screenshot from 2024-07-01 11-04-37](https://github.com/openshift/openshift-docs/assets/57954076/ea6eb532-472e-4aa1-8f20-de3dee75649c)